Bug: runserver --with-pages wasm-bindgen uses wrong target path in workspace setups

## Prerequisites

- [x] I have searched existing issues to verify this bug hasn't been reported

## Bug Description

When `runserver --with-pages` is executed from a workspace member crate directory (e.g., `cd dashboard && cargo run --bin manage -- runserver --with-pages`), the `wasm-bindgen` step looks for the WASM file at `./target/wasm32-unknown-unknown/debug/<crate>.wasm` relative to the current working directory. However, cargo builds to the workspace root's `target/` directory.

### Expected behavior

`wasm-bindgen` should resolve the WASM artifact path relative to the workspace root (using `cargo metadata` to find the workspace root and target directory).

### Actual behavior

```
[ERROR] WASM build failed: wasm-bindgen failed: error: failed reading './target/wasm32-unknown-unknown/debug/reinhardt_cloud_dashboard.wasm'
Caused by:
    No such file or directory (os error 2)
```

The file actually exists at `<workspace_root>/target/wasm32-unknown-unknown/debug/reinhardt_cloud_dashboard.wasm`.

### Steps to reproduce

1. Have a cargo workspace with a member crate that uses `pages` feature
2. `cd <member_crate>`
3. `cargo run --bin manage -- runserver --with-pages`
4. Observe wasm-bindgen failure

### Note

Running from the workspace root doesn't trigger the build at all (likely because `src/` is not found). The fix should handle both cases.

### Related

Follows up on PR #3292 which added the `--lib` flag to the WASM cargo build step.

### Environment

- reinhardt-web: b2f14908 (main branch)
- Rust: stable
- OS: macOS

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/reinhardt-commands/src/bin/runserver.rs
+++ b/crates/reinhardt-commands/src/bin/runserver.rs
@@ -551,7 +551,19 @@ fn build_pages_wasm(force: bool) -> bool {
 		"{}",
 		format!("Building pages WASM for {}...", crate_name).cyan()
 	);
-	let config = WasmBuildConfig::new(".").output_dir("dist");
+	// Resolve workspace root so wasm-bindgen finds the artifact in the
+	// workspace-level target directory, not relative to the member crate CWD.
+	let manifest_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+	let workspace_root = manifest_dir
+		.parent()
+		.and_then(|p| p.parent())
+		.and_then(|p| p.parent())
+		.and_then(|p| p.parent())
+		.map(PathBuf::from)
+		.unwrap_or_else(|| PathBuf::from("."));
+	let config = WasmBuildConfig::new(".")
+		.output_dir("dist")
+		.target_dir(workspace_root.join("target"));
 	match WasmBuilder::new(config).build() {
 		Ok(_) => {
 			println!("{}", "Pages WASM build succeeded.".green());

--- a/crates/reinhardt-commands/src/wasm_builder.rs
+++ b/crates/reinhardt-commands/src/wasm_builder.rs
@@ -20,6 +20,10 @@ pub struct WasmBuildConfig {
 	pub optimize: bool,
 	/// Target name (crate name, used for output file naming)
 	pub target_name: Option<String>,
+	/// Override for the cargo target directory. When `None`, falls back to
+	/// `project_dir/target`. In workspace setups, this should point to the
+	/// workspace root's target directory.
+	pub target_dir: Option<PathBuf>,
 }
 
 impl Default for WasmBuildConfig {
@@ -30,6 +34,7 @@ impl Default for WasmBuildConfig {
 			release: false,
 			optimize: true,
 			target_name: None,
+			target_dir: None,
 		}
 	}
 }
@@ -64,6 +69,16 @@ impl WasmBuildConfig {
 	/// Set the target name explicitly.
 	pub fn target_name(mut self, name: impl Into<String>) -> Self {
 		self.target_name = Some(name.into());
+		self
+	}
+
+	/// Set the cargo target directory explicitly.
+	///
+	/// When building inside a Cargo workspace, the target directory is at
+	/// the workspace root, not relative to the member crate. Use this to
+	/// point wasm-bindgen at the correct artifact location.
+	pub fn target_dir(mut self, dir: impl Into<PathBuf>) -> Self {
+		self.target_dir = Some(dir.into());
 		self
 	}
 }
@@ -146,10 +161,13 @@ impl WasmBuilder {
 		} else {
 			"debug"
 		};
-		let wasm_path = self
+		let target_base = self
 			.config
-			.project_dir
-			.join("target")
+			.target_dir
+			.as_ref()
+			.cloned()
+			.unwrap_or_else(|| self.config.project_dir.join("target"));
+		let wasm_path = target_base
 			.join("wasm32-unknown-unknown")
 			.join(profile)
 			.join(format!("{}.wasm", crate_name.replace('-', "_")));

